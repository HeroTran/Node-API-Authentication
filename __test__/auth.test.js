const request = require('supertest')
const app = require('../app')
describe('Post Endpoints', () => {
  it('should create a new user', async () => {
    const user = {
      "name": "herotran210",
      "email": "herotran1" + Math.floor((Math.random() * 10000) + 1) + "@gmail.com",
      "password": "123456"
    }
    const res = await request(app)
      .post('/api/user/register')
      .send(user)
    expect(res.statusCode).toEqual(201)
    expect(res.body.status).toEqual(true)
  })
  it('returns all users', async () => {
    const response = await request(app).get('/api/user/list');
    expect(response.body).toMatchSnapshot();
  });
})