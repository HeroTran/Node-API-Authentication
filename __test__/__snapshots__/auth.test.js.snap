// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Endpoints returns all users 1`] = `
Array [
  Object {
    "__v": 0,
    "_id": "5dca6e63e0d47a65009738e0",
    "date": "2019-11-12T08:33:37.080Z",
    "email": "herotran1@gmail.com",
    "name": "herotran210",
    "password": "$2b$10$Gw8.evQvmn29ta6lCuM7LusHw23qVClUPip7xA3/aCjEIpvzjEh2C",
  },
  Object {
    "__v": 0,
    "_id": "5dca6f0c540dbf6038f6fa12",
    "date": "2019-11-12T08:36:25.854Z",
    "email": "herotran15106@gmail.com",
    "name": "herotran210",
    "password": "$2b$10$Vi/VNNeJlCJBWrqp30GkmOca2MkUF.E.f8Ene98JAqx0.yDiv3bfS",
  },
]
`;
